cmd/fluxd/main.go

```diff
@@ -308,6 +310,18 @@ func main() {
 		*automationInterval = *registryPollInterval
 	}
 
+	// Maintain backwards compatibility with the --git-verify-signatures
+	// flag, but only if the --git-verify-signature-mode is not set to a
+	// custom (non default) value.
+	var gitVerifySignaturesMode fluxsync.VerifySignaturesMode
+	if *gitVerifySignatures && !fs.Changed("git-verify-signatures-mode") {
+		gitVerifySignaturesMode = fluxsync.VerifySignaturesModeAll
+	} else {
+		if gitVerifySignaturesMode, err = fluxsync.ToVerifySignaturesMode(*gitVerifySignaturesModeStr); err != nil {
+			logger.Log("error", err.Error())
```

Comment:
Having doubts about the fact that this does not make Flux exit given this all is a safety feature. I think it may be better to:

```suggestion
			logger.Log("error", err.Error())
			os.Exit(1)
```

to prevent things from getting applied without validation, as changing the default value automatically means that you want some type of validation to happen.